signer/signer.go

```diff
@@ -133,3 +144,27 @@ func (s *genericSigner) Sign(ctx context.Context, desc ocispec.Descriptor, opts
 	// TODO: re-enable timestamping https://github.com/notaryproject/notation-go/issues/78
 	return sig, &envContent.SignerInfo, nil
 }
+
+func constructPayload(ctx context.Context, desc ocispec.Descriptor, userMetadata map[string]string) (*envelope.Payload, error) {
+	logger := log.GetLogger(ctx)
+
+	if desc.Annotations == nil && len(userMetadata) > 0 {
+		desc.Annotations = map[string]string{}
+	}
```

Comment:
Why are we overriding existing annotations instead we should append new ones?

Reply:
this is initializing annotations to an empty map if the existing descriptor has a nil annotation map to avoid trying to write to a nil map.